hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/JetPlanExecutor.java

```diff
@@ -246,4 +244,14 @@ SqlResult execute(SelectOrSinkPlan plan, QueryId queryId, List<Object> arguments
 
         return arguments;
     }
+
+    private static int findQueryExceptionCode(Throwable t) {
+        while (t != null) {
+            if (t instanceof QueryException) {
+                return ((QueryException) t).getCode();
+            }
+            t = t.getCause() == t ? null : t.getCause();
```

Comment:
This looks like a hack. `Throwable.getCause()` returns `null`, if `cause == this`. We rely on this elsewhere already, e.g. in  Do we need to do this to workaround some exception that returns `this` from `getCause`? There still can be cycles, e.g. `t.getCause().initCause(t)`.